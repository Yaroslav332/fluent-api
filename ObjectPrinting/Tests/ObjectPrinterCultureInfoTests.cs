using System.Globalization;
using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ObjectPrinting.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class ObjectPrinterCultureInfoTests
    {
        [Test]
        public void SetCulture_Should_ChangeFormat()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo = new CultureInfo("ru");
            var printer = ObjectPrinter.For<Person>()
                .SetCulture(cultureInfo);
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void SetCulture_Should_OverridePreviousSetCulture()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo1 = new CultureInfo("ru");
            var cultureInfo2 = new CultureInfo("en");
            var printer = ObjectPrinter.For<Person>()
                .SetCulture(cultureInfo1)
                .SetCulture(cultureInfo2);
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void SetCulture_ShouldNot_OverridePreviousMethod()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo = new CultureInfo("ru");
            var printer = ObjectPrinter.For<Person>()
                .ConfigForProperty(p => p.Height)
                .UseSerializeMethod(f => "123haha")
                .SetCulture(cultureInfo);
            Approvals.Verify(printer.PrintToString(p));
        }

        [Test]
        public void UseSerializeMethod_Should_OverridePreviousSetCulture()
        {
            var p = new Person {Name = "Alex", Height = 35.5, Weight = 83.3};
            var cultureInfo = new CultureInfo("ru");
            var printer = ObjectPrinter.For<Person>()
                .SetCulture(cultureInfo)
                .ConfigForProperty(p => p.Height)
                .UseSerializeMethod(f => "123haha");
            Approvals.Verify(printer.PrintToString(p));
        }
    }
}