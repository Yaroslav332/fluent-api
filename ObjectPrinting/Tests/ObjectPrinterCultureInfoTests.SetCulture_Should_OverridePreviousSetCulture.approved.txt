Operation
	Operator1 = 0.1
	Operator2 = 0.2
	Operator3 = 0.3
	Operator4 = 0.4
	Operator5 = 0.5
	Operator6 = 0.6
	OperationDate = 5/3/1994 12:00:00 AM
